rmw_fastrtps_shared_cpp/src/rmw_wait.cpp

```diff
@@ -100,8 +101,13 @@ __rmw_wait(
 {
   if (!wait_set) {
     RMW_SET_ERROR_MSG("wait set handle is null");
-    return RMW_RET_ERROR;
+    return RMW_RET_INVALID_ARGUMENT;
   }
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    wait set handle,
+    wait_set->implementation_identifier, identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION)
+
   CustomWaitsetInfo * wait_set_info = static_cast<CustomWaitsetInfo *>(wait_set->data);
   if (!wait_set_info) {
```

Comment:
@ahcorde I think we can remove this check. If `wait_set_info` is ever `nullptr`, it can only mean one of three things:

1. Wait set is invalid. Caller did not respect preconditions. We cannot check for pointer validity in general, checking for `nullptr` is near useless. 
2. Implementation is logically broken. Definitely not something we want to treat as a normal error.
3. Heap is corrupt.

In all three cases, it's better if this crashes soon enough.

Reply:
On that same note, [lines 118 to 125](https://github.com/ros2/rmw_fastrtps/pull/443/files#diff-6c98126e644ee46f5d86c7fffa3da911R118-R125) are nonsense. Those cannot ever be `nullptr` (see [CustomWaitSetInfo](https://github.com/ros2/rmw_fastrtps/blob/8f6a13c5995b13a8bd9a0d147851ab5754bdf104/rmw_fastrtps_shared_cpp/src/types/custom_wait_set_info.hpp#L21)).